### PR TITLE
Print extra info after main error message, don't treat backtrace specially

### PR DESCRIPTION
**Pull request by SkySkimmer**
IMO it's nicer to get "error messsage; ltac2 backtrace" than "error ltac2 backtrace; message"

### CODE REVIEW

**Review by ejgallego — APPROVED** [review]
Long overdue IMO, thanks!

@SkySkimmer do you think this need more review from the ltac maintainers?

**Comment by ejgallego** [comment]
@SkySkimmer can you rebase so we get a CI run? @ppedrot wanna have a look at this?

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by herbelin** [comment]
Maybe what was previously missing was a good articulation between the two informations. With the new layout, I feel that something is still missing. What about:
```coq
The term "I" has type "True" while it is expected to have type "False".
This error was raised by the last call from the following sequence of nested calls: "g2 (constr)", "g1" and "refine (uconstr)".
```
or something like that?

**Review by ppedrot — APPROVED** [review]
LGTM.

**Comment by ejgallego** [comment]
@SkySkimmer would you like to address Hugo's comment, or shall I merge as-is?

**Comment by SkySkimmer** [comment]
I think this PR can be merged asis

**Pull request merged**
